rllib/examples/curriculum_learning.py

```diff
@@ -0,0 +1,99 @@
+"""
+Example of a curriculum learning setup using the `TaskSettableEnv` API
+and the env_task_fn config.
+
+This example shows:
+  - Writing your own curriculum-capable environment using gym.Env.
+  - Defining a env_task_fn that determines, whether and which new task
+    the env(s) should be set to (using the TaskSettableEnv API).
+  - Using Tune and RLlib to curriculum-learn this env.
+
+You can visualize experiment results in ~/ray_results using TensorBoard.
+"""
+import argparse
+import numpy as np
+import os
+
+import ray
+from ray import tune
+from ray.rllib.env.apis.task_settable_env import TaskSettableEnv, TaskType
+from ray.rllib.env.env_context import EnvContext
+from ray.rllib.examples.env.curriculum_capable_env import CurriculumCapableEnv
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.test_utils import check_learning_achieved
+
+tf1, tf, tfv = try_import_tf()
+torch, nn = try_import_torch()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--run", type=str, default="PPO")
+parser.add_argument("--torch", action="store_true")
+parser.add_argument("--as-test", action="store_true")
+parser.add_argument("--stop-iters", type=int, default=50)
+parser.add_argument("--stop-timesteps", type=int, default=200000)
+parser.add_argument("--stop-reward", type=float, default=10000.0)
```

Comment:
add helpstring?

Reply:
Done. I'll do this for all the other example scripts in a separate PR.